tests/test_api.py

```diff
@@ -0,0 +1,332 @@
+import asyncio
+import logging
+import os
+from datetime import datetime, timedelta, timezone
+from decimal import Decimal
+from typing import Any
+
+import grpc
+import pytest
+
+from frequenz.client.electricity_trading import (
+    Client,
+    Currency,
+    DeliveryArea,
+    DeliveryPeriod,
+    Energy,
+    EnergyMarketCodeType,
+    MarketSide,
+    OrderDetail,
+    OrderState,
+    OrderType,
+    Price,
+)
+
+# Setup logging configuration
+logging.basicConfig(
+    level=logging.DEBUG, format="%(asctime)s - %(name)s - %(levelname)s - %(message)s"
+)
+_logger = logging.getLogger(__name__)
+
+GRIDPOOL_ID = 6
+API_KEY = os.getenv("API_KEY", None)
+if not API_KEY:
+    raise ValueError("API Key is not set")
+
+
+@pytest.fixture
+async def set_up() -> dict[str, Any]:
+    """Set up the test suite."""
+    client = Client(
+        server_url="grpc://electricity-trading-testing.api.frequenz.com:443?ssl=true",
+        auth_key=API_KEY,
+    )
+
+    delivery_area = DeliveryArea(
+        code="10YDE-EON------1", code_type=EnergyMarketCodeType.EUROPE_EIC
+    )
+    # Setting delivery start to 3hours from now
+    delivery_start = (datetime.now(timezone.utc) + timedelta(hours=3)).replace(
+        minute=0, second=0, microsecond=0
+    )
+    delivery_period = DeliveryPeriod(
+        start=delivery_start,
+        duration=timedelta(minutes=15),
+    )
+    price = Price(amount=Decimal("56"), currency=Currency.EUR)
+    quantity = Energy(mwh=Decimal("0.1"))
+    order_type = OrderType.LIMIT
+
+    return {
+        "client": client,
+        "delivery_area": delivery_area,
+        "delivery_period": delivery_period,
+        "price": price,
+        "quantity": quantity,
+        "order_type": order_type,
+    }
+
+
+async def create_test_order(
+    set_up: dict[str, Any],
+    side: MarketSide = MarketSide.BUY,
+    price: Price | None = None,
+    delivery_period: DeliveryPeriod | None = None,
+) -> OrderDetail:
+    """Create a test order with customizable parameters."""
+    order_price = price or set_up["price"]
+    order_delivery_period = delivery_period or set_up["delivery_period"]
+    order = await set_up["client"].create_gridpool_order(
+        gridpool_id=GRIDPOOL_ID,
+        delivery_area=set_up["delivery_area"],
+        delivery_period=order_delivery_period,
+        order_type=set_up["order_type"],
+        side=side,
+        price=order_price,
+        quantity=set_up["quantity"],
+    )
+    return order  # type: ignore
+
+
+async def create_test_trade(
+    set_up: dict[str, Any],
+) -> tuple[OrderDetail, OrderDetail]:
+    """
+    Create identical orders on opposite sides to try to trigger a trade.
+
+    Args:
+        set_up: The setup dictionary.
+    Returns:
+        A tuple of the created buy and sell orders.
+    """
+    # Set a different delivery period so that it is the only trade retrieved
+    delivery_start = (datetime.now(timezone.utc) + timedelta(hours=4)).replace(
```

Comment:
yea it is, but does it make much of a difference?

Reply:
What if someone changes the setup delivery start? Also gives you an idea of what is happening without having to look up setup. But that's just a suggestion

Reply:
Oh I see, yep good call but I forgot to also mention that the trade `delivery_start` should be always < 9 hours away from now (since the order book for the next day opens at 15:00). So in that case if someone changes the set_up delivery_start there is a risk that it isn't matched. Maybe I add this to the comment tho so it's clear